lexer.js

```diff
@@ -651,8 +651,10 @@ function tryParseObjectDefineOrKeys (keys) {
             if (ch !== 33/*!*/) break;
             pos += 1;
             ch = commentWhitespace();
-            if (source.startsWith(id, pos)) {
-              pos += id.length;
+            if (ch === 79/*O*/ && source.startsWith('bject', pos + 1) && source[pos + 6] === '.') {
+              if (!tryParseObjectHasOwnProperty(it_id)) break;
+            }
```

Comment:
This has just been moved here from the old `else` branch without changing the semantics, right?

Reply:
Yes, the order switch is so that `pos` doesn't get changed for the fallback check.